File: csc330/poundsPractice/Machine.java
public class Machine {

   public static void main( String[] args) {


// Use the different constructors created in the Drink class.       
    Drink drink1 = new Drink("Coke");
    Drink drink2 = new Drink("Sprite");
  //Drink drink3 = new Drink("Doctor Pepper"); 
  //Replace Dr. Pepper with Blue Powerade for $2.75
    Drink drink3 = new Drink("Blue Powerade", 2.75);
    Drink drink4 = new Drink("Guinness",6.25);

// Fill all of your drinks here  

    drink1.refill();
    drink2.refill();
    drink3.refill();
    drink4.refill();

// Purchase 8 Cokes here

    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();

// Purchase 6 Sprites here7

    drink2.vend();
    drink2.vend();
    drink2.vend();
    drink2.vend();
    drink2.vend();
    drink2.vend();

// Purchase 2 Doctor Peppers here

    drink3.vend();
    drink3.vend();


// Purchase 11 Guinness here

    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();


// Compute and print out the total number of drinks sold and the total
// profit you made

    int totalSold = drink1.drinksSold() + drink2.drinksSold() +
                    drink3.drinksSold() + drink4.drinksSold();

    double totalProfit = drink1.profit() + drink2.profit() +
                         drink3.profit() + drink4.profit();

    System.out.print("A Total of " + totalSold + " drinks were ");
    System.out.println("sold for a profit of $" + totalProfit);
    System.out.println();
    System.out.println( drink1.restockAmount() + " " + drink1.getDrinkName() + " needed to restock machine");
    System.out.println( drink2.restockAmount() + " " + drink2.getDrinkName() + " needed to restock machine");
    System.out.println( drink3.restockAmount() + " " + drink3.getDrinkName() + " needed to restock machine");
    System.out.println( drink4.restockAmount() + " " + drink4.getDrinkName() + " needed to restock machine");

      }
}

File: csc330/poundsPractice/juliaPractice/Machine.jl
include("Drink.jl")

drink = Vector{Drink}(undef,4)

# Create Drink vector
drink[1] = Drink("Coca-Cola")
drink[2] = Drink("Sprite")
drink[3] = Drink("Guinness", 6.50)
drink[4] = Drink("Bud Light", 0.25)

# Refill the drink machine.
for i in 1:4
  refill(drink[i])
end

# Buy 4 Cokes.
for i in 1:4
vend(drink[1])
end

# Buy 8 Sprites.
for i in 1:8
vend(drink[2])
end

# Buy 11 Guinness
for i in 1:11
  vend(drink[3])
end

# Buy 1 Bud Light
for i in 1:1
  vend(drink[4])
end

# Refill Guinness
refill(drink[3])

# Compute the total Profit
profit = 0.0
for i in 1:4
  global profit += getProfit(drink[i])
end

# Show the total profit
println("The total profit is \$" * string(profit))

# Generate a report per drink

for i in 1:4
  println( "You sold " * string(getDrinksSold(drink[i])) * " " * getDrinkName(drink[i]) * " for a total of \$" * string(getProfit(drink[i])) )
  println( "         " * string(getRestockAmount(drink[i])) * " of " * getDrinkName(drink[i]) * " needed to restock the machine " )
end
